pop11_grammar_ebnf.txt

```diff
@@ -83,25 +87,25 @@ Jump ::=
 	'goto' Word
 
 Iterative ::=
-	'for' Expression 'step' Expression 'till' Expression 'do' Sequence 'endfor'
-	| 'while' Expression 'do' Sequence 'endwhile'
-	| 'until' Expression 'do' Sequence 'enduntil'
-	| 'for' Word (('in' | 'on') | ('from' Expression ('by' Expression)?)? 'to') Expression 'do' Sequence 'endfor'
-	| 'for' Word 'from_repeater' Expression 'do' Sequence 'endfor'
-	| 'foreach' Expression ('in' Expression)? 'do' Sequence 'endforeach'
-	| 'forevery' Expression ('in' Expression)? 'do' Sequence 'endforevery'
-	| 'repeat' (Expression 'times' | 'forever')? Sequence 'endrepeat'
+	'for' NonEmptyExpression 'step' NonEmptyExpression 'till' NonEmptyExpression 'do' Sequence 'endfor'
+	| 'while' NonEmptyExpression 'do' Sequence 'endwhile'
+	| 'until' NonEmptyExpression 'do' Sequence 'enduntil'
+	| 'for' Word (('in' | 'on') | ('from' NonEmptyExpression ('by' NonEmptyExpression)?)? 'to') NonEmptyExpression 'do' Sequence 'endfor'
+	| 'for' Word 'from_repeater' NonEmptyExpression 'do' Sequence 'endfor'
+	| 'foreach' NonEmptyExpression ('in' NonEmptyExpression)? 'do' Sequence 'endforeach'
+	| 'forevery' NonEmptyExpression ('in' NonEmptyExpression)? 'do' Sequence 'endforevery'
+	| 'repeat' (NonEmptyExpression 'times' | 'forever')? Sequence 'endrepeat'
 
 Conditional ::=
-	'if' Expression 'then' Sequence (('elseif' | 'elseunless') Expression 'then' Sequence)* ('else' Expression)? 'endif'
-	| 'unless' Expression 'then' Sequence (('elseif' | 'elseunless') Expression 'then' Sequence)* ('else' Expression)? 'endunless'
+	'if' NonEmptyExpression 'then' Sequence (('elseif' | 'elseunless') NonEmptyExpression 'then' Sequence)* ('else' NonEmptyExpression)? 'endif'
+	| 'unless' NonEmptyExpression 'then' Sequence (('elseif' | 'elseunless') NonEmptyExpression 'then' Sequence)* ('else' NonEmptyExpression)? 'endunless'
 
 BooleanExpression ::=
-	Expression (('and' | 'or') Expression)*
+	NonEmptyExpression (('and' | 'or') NonEmptyExpression)*
 
 OperatorExpression ::=
-	Expression Operator Expression
+	NonEmptyExpression Operator NonEmptyExpression
 
 FunctionApplication ::=
-	Expression '(' Expression? (',' Expression?)* ')'
-	| '.' Expression
+	NonEmptyExpression '(' Expression (',' Expression)* ')'
+	| '.' NonEmptyExpression

```

Comment:
Two errors here - one I previously failed to spot. It should be: `Expression '.' Expression`. Both LHS & RHS may be empty, insane as that might seem. Example:
```
 Setpop
 : identfn; .; =>
 **
 : 
```